synapseclient/models/file.py

```diff
@@ -134,14 +295,97 @@ class File(AccessControllable):
     """Additional metadata associated with the folder. The key is the name of your
     desired annotations. The value is an object containing a list of values
     (use empty list to represent no values for key) and the value type associated with
-    all values in the list."""
+    all values in the list. To remove all annotations set this to an empty dict `{}`."""
+
+    create_or_update: bool = field(default=True, repr=False)
+    """
+    (Store only)
+
+    Indicates whether the method should automatically perform an update if the file
+    conflicts with an existing Synapse object.
+    """
+
+    force_version: bool = field(default=True, repr=False)
+    """
+    (Store only)
+
+    Indicates whether the method should increment the version of the object even if
+    nothing has changed.
+    """
+
+    is_restricted: bool = field(default=False, repr=False)
+    """
+    (Store only)
+
+    If set to true, an email will be sent to the Synapse access control team to start the
+    process of adding terms-of-use or review board approval for this entity.
+    You will be contacted with regards to the specific data being restricted and the
+    requirements of access.
+    """
+
+    synapse_store: bool = field(default=True, repr=False)
+    """
+    (Store only)
+
+    Whether the File should be uploaded or if only the path should be stored when
+    [synapseclient.models.File.store][] is called.
+    """
+
+    download_file: bool = field(default=True, repr=False)
+    """
+    (Get only)
+
+    If True the file will be downloaded."""
+
+    download_location: str = field(default=None, repr=False)
+    """
+    (Get only)
+
+    The location to download the file to."""
 
-    # TODO: We need to provide functionality for folks to store the file in Synapse, but
-    # TODO: Not upload the file.
+    if_collision: str = field(default="keep.both", repr=False)
+    """
+    (Get only)
+
+    Determines how to handle file collisions. Defaults to "keep.both".
+            May be
+
+            - `overwrite.local`
+            - `keep.local`
+            - `keep.both`
+    """
+
+    synapse_container_limit: Optional[str] = field(default=None, repr=False)
+    """A Synanpse ID used to limit the search in Synapse if file is specified as a local
+    file. That is, if the file is stored in multiple locations in Synapse only the
+    ones in the specified folder/project will be returned."""
+
+    _last_persistent_instance: Optional["File"] = field(default=None, repr=False)
+    """The last persistent instance of this object. This is used to determine if the
+    object has been changed and needs to be updated in Synapse."""
+
+    def _set_last_persistent_instance(self) -> None:
```

Comment:
Not in this case - This is used to know if you modified aspects of the data in between the last time you did a (store, get) and the next time you do it and you pass along the same instance of the File object.

Currently I am just checking if annotations or Activity has changed, and saving of those values if it's non-None and has changed. Since storing of the Entity data (POST `/entity/{id}`) is done within the `client.py` I currently cannot check if there is a no-op in this case. However, I am thinking that when we move this to be purely async code and we pull that logic out of `client.py` we can include a `has_changed` check there for the file level attributes too.